src/nav/nav.ts

```diff
@@ -0,0 +1,204 @@
+import {
+  AfterContentChecked,
+  ChangeDetectorRef,
+  ContentChildren,
+  Directive,
+  EventEmitter,
+  forwardRef,
+  Inject,
+  Input,
+  Output,
+  QueryList,
+  TemplateRef
+} from '@angular/core';
+import {isDefined} from '../util/util';
+import {NgbNavConfig} from './nav-config';
+
+let navCounter = 0;
+
+/**
+ * This directive must be used to wrap content to be displayed in the nav.
+ */
+@Directive({selector: 'ng-template[ngbNavContent]'})
+export class NgbNavContent {
+  constructor(public templateRef: TemplateRef<any>) {}
+}
+
+
+/**
+ * Nav item
+ */
+@Directive({selector: '[ngbNavItem]', exportAs: 'ngbNavItem', host: {'class': 'nav-item'}})
+export class NgbNavItemDirective implements AfterContentChecked {
+  private _defaultDomId = `ngb-nav-${navCounter++}`;
+  private _nav: NgbNavDirective;
+
+  /**
+   * If `true`, non-active current nav item content will be removed from DOM
+   * Otherwise it will just be hidden
+   */
+  @Input() destroyOnHide;
+
+  /**
+   * If `true`, the current nav item is disabled and can't be toggled.
+   */
+  @Input() disabled = false;
+
+  /**
+   * The id used for the DOM elements.
+   * Must be unique inside the document in case you have multiple `ngbNav`s on the page.
+   *
+   * Autogenerated as `ngb-nav-XXX` if not provided.
+   */
+  @Input('domId') _domId: string;
+
+  /**
+   * The id used as a model for active nav.
+   * It can be anything, but must be unique inside one `ngbNav`.
+   */
+  @Input('ngbNavItem') _id: any;
+
+  contentTpl: NgbNavContent | null;
+
+  @ContentChildren(NgbNavContent, {descendants: false}) contentTpls: QueryList<NgbNavContent>;
+
+  constructor(@Inject(forwardRef(() => NgbNavDirective)) nav) {
+    // TODO: cf https://github.com/angular/angular/issues/30106
+    this._nav = nav;
+  }
+
+  ngAfterContentChecked() {
+    // We are using @ContentChildren instead of @ContentChild as in the Angular version being used
+    // only @ContentChildren allows us to specify the {descendants: false} option.
+    // Without {descendants: false} we are hitting bugs described in:
+    // https://github.com/ng-bootstrap/ng-bootstrap/issues/2240
+    this.contentTpl = this.contentTpls.first;
+  }
+  get active() { return this._nav.activeId === this.id; }
+
+  get domId() { return this._domId || this._defaultDomId; }
```

Comment:
Maybe we can remove the usage of _defaultDomId. Just set _domId if it's equal to null. 

It will simplify the code and avoid the `||` usage each time domId  is invoked.

Reply:
done!